src/Features/CSharp/Portable/QuickInfo/CSharpDiagnosticAnalyzerQuickInfoProvider.cs

```diff
@@ -36,13 +31,16 @@ public CSharpDiagnosticAnalyzerQuickInfoProvider(IDiagnosticAnalyzerService diag
             _diagnosticAnalyzerService = diagnosticAnalyzerService;
         }
 
-        protected override async Task<QuickInfoItem?> BuildQuickInfoAsync(
-            Document document,
-            SyntaxToken token,
-            CancellationToken cancellationToken)
+        protected override async Task<QuickInfoItem?> BuildQuickInfoAsync(QuickInfoContext context)
         {
-            return GetQuickinfoForPragmaWarning(document, token) ??
-                (await GetQuickInfoForSuppressMessageAttributeAsync(document, token, cancellationToken).ConfigureAwait(false));
+            if (context.Document is { } document)
```

Comment:
i don't think `context.Document` could ever be null. that would be *super* weird for this domain.

Reply:
oh, is that effectively what you're trying to do here?  make either one or the other optional?  interesting...

Reply:
Yes, eventually we probably want separate interfaces - the one exposed to public will always have non-null Document and only work in document based quick info queries and the one that is internal for our providers will always have non-null SemanticModel but can have null Document (for LSIF scenario). I think we can do this as a follow-up change.